### PR TITLE
feat: connect alby account nwc node

### PR DESCRIPTION
**Pull request by rolznz**
fixes https://github.com/getAlby/nostr-wallet-connect-next/issues/194

This PR automatically links the user's Alby Account to the Alby Hub to the Alby account (and creating a NWC connection for getalby.com) the first time the user authenticates with Alby via OAuth.

Also adds unstyled success page (https://github.com/getAlby/nostr-wallet-connect-next/issues/234)

image

### CODE REVIEW

**Comment by rolznz** [comment]
Update: do this automatically once the user connects to their Alby account for the first time. Do not add a new page the user has to read through. CC @reneaaron

This will get users stuck if they don't finish the flow or a channel is pending, which they can disconnect from the getalby.com UI if they really want to. But that the happy day scenario works great

**Comment by rolznz** [comment]
@reneaaron @bumi I updated the PR so now it automatically links the first the OAuth callback happens.

**Comment by rolznz** [comment]
Now there is some coupling between the Alby OAuth service and the service, since the Alby OAuth Service needs an API to be able to create the app for the getalby.com NWC connection.

Related issue: https://github.com/getAlby/nostr-wallet-connect-next/issues/17

File: alby/alby.go [hunk 1]
```diff
@@ -332,6 +350,39 @@ func (svc *AlbyOAuthService) GetAuthUrl() string {
...
+	connectionPubkey, err := svc.createAlbyAccountNWCNode(ctx)
+	if err != nil {
+		svc.logger.WithError(err).Error("Failed to create alby account nwc node")
+		return err
+	}
+
+	app, err := svc.api.CreateApp(&api.CreateAppRequest{
+		Name:           "getalby.com",
+		Pubkey:         connectionPubkey,
+		MaxAmount:      100_000,
```
Comment (bumi) [inline alby/alby.go:363]: this should be higher. as long as we don't have a good flow to update the budget. 
make it a million? 
it's a well behaving app and likely one bigger spending apps.

File: alby/alby.go [hunk 2]
```diff
@@ -65,35 +68,50 @@ func NewAlbyOauthService(logger *logrus.Logger, kvStore config.ConfigKVStore, ap
...
 	if err != nil {
-		svc.logger.WithError(err).Error("Failed to fetch user me")
+		svc.logger.WithError(err).Error("Failed to exchange token")
 		return err
 	}
+	svc.saveToken(token)
 
-	svc.kvStore.SetUpdate(userIdentifierKey, me.Identifier, "")
+	// setup Alby account on first time login
+	if existingAccessToken == "" {
```
[the remark this answers is not in the document]
Comment (rolznz) [reply alby/alby.go]: Sounds good to me, thanks

File: alby/alby.go (hunk 1 above)
Comment (rolznz) [reply alby/alby.go]: sounds good, done

**Pull request merged**